glean-core/csharp/csharp/Glean.cs

```diff
@@ -0,0 +1,177 @@
+﻿// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+using Mozilla.Glean.FFI;
+using System;
+using System.IO;
+
+namespace Mozilla.Glean
+{
+    /// <summary>
+    /// The Glean Gneral API.
+    /// </summary>
+    public class Glean
+    {
+        // Initialize the singleton using the `Lazy` facilities.
+        private static readonly Lazy<Glean>
+          lazy = new Lazy<Glean>(() => new Glean());
+        public static Glean GleanInstance => lazy.Value;
+
+        private Glean()
+        {
+            // Private constructor, intentionally mostly empty, to
+            // disallow instantiation since this is meant to
+            // be a singleton.
+            LibGleanFFI.glean_enable_logging();
+        }
+
+        /// <summary>
+        /// Initialize the Glean SDK.
+        /// 
+        /// This should only be initialized once by the application, and not by
+        /// libraries using the Glean SDK. A message is logged to error and no
+        /// changes are made to the state if initialize is called a more than
+        /// once.
+        /// 
+        /// A LifecycleObserver will be added to send pings when the application goes
+        /// into foreground and background.
```

Comment:
I realize this is a rough framework, but since we don't know this yet, maybe it's a good idea to remove this now before it gets left behind, or maybe update it with a TODO or something?